src/Support/SelectFields.php

```diff
@@ -294,6 +295,26 @@ private static function isQueryable(array $fieldObject): bool
         return Arr::get($fieldObject, 'is_relation', true) === true;
     }
 
+    /**
+     * Retrieves the SQL column alias expression.
+     *
+     * @param $fieldObject
+     *
+     * @return Expression | null | string
+     */
+    private static function getColumn(string $key, $fieldObject)
+    {
+        if (isset($fieldObject->config['alias'])) {
+            $alias = $fieldObject->config['alias'];
+
+            if (is_string($alias)) {
+                return $alias.' AS '.$key;
+            } elseif (is_callable($alias)) {
+                return DB::raw('('.$alias().') AS '.$key);
```

Comment:
I may be wrong but generating a fragment like `DB::raw('('.$alias().') AS '.$key)` will not properly fully quality `$key` and there are databases like PostgreSQL which would need wrapping in `"` if the name contains upper case characters. This is just one case which comes to my mind.

Reply:
your solution seems more straith forward, even though it is a bit more work, so closing this